src/om_bootstrap/nav.cljs

```diff
@@ -169,15 +169,21 @@
       (u/clone-with-props child prop-fn))))
 
 (defn render-navbar-child [owner child bs]
-  (let [f (fn [props]
+  (let [changing? (om/get-state owner :changing?)
+        f (fn [props]
             (let [opts (:opts props)
                   collapsible? (when (:toggle-nav-key bs)
                                  (= (:key opts) (:toggle-nav-key bs)))
+                  expanded? (and collapsible?
+                                 (or (:nav-expanded? bs)
+                                     (om/get-state owner :nav-open?)))
+                  collapsed? (not expanded?)
                   base {:navbar? true
-                        :collapsible? collapsible?
-                        :expanded? (and collapsible?
-                                        (or (:nav-expanded? bs)
-                                            (om/get-state owner :nav-open?)))}]
+                        :class (d/class-set {:navbar-collapse collapsible?
```

Comment:
You want to respect encapsulation by only setting the properties supplied to the `Nav` class:

```
(def Nav
  (t/bootstrap
   {:bs-style (s/enum "tabs" "pills")
    (s/optional-key :active-key) (s/either s/Str s/Num)
    (s/optional-key :active-href) s/Str
    (s/optional-key :stacked?) s/Bool
    (s/optional-key :justified?) s/Bool
    (s/optional-key :collapsible?) s/Bool
    (s/optional-key :expanded?) s/Bool
    (s/optional-key :navbar?) s/Bool
    (s/optional-key :pull-right?) s/Bool}))
```
and letting it set its own class internally, instead of jumping inside the API. Is there a reason you can't move this `class-set` change inside of the `Nav`'s render function?